src/esp/core/Configuration.cpp

```diff
@@ -183,6 +184,19 @@ std::string ConfigValue::getAsString() const {
       auto v = get<Mn::Vector3>();
       return Cr::Utility::formatString("[{} {} {}]", v.x(), v.y(), v.z());
     }
+    case ConfigStoredType::MagnumMat3: {
+      auto m = get<Mn::Matrix3>();
+      auto v = m.row(0);
+      std::string res =
+          Cr::Utility::formatString("[[{} {} {}]", v.x(), v.y(), v.z());
+      for (int i = 1; i < m.Size; ++i) {
+        v = m.row(i);
+        Cr::Utility::formatInto(res, res.length(), ", [{} {} {}]", v.x(), v.y(),
+                                v.z());
+      }
+      Cr::Utility::formatInto(res, res.length(), "]");
+      return res;
+    }
```

Comment:
Ah wait, why is this class reinventing Magnum's own builtin debug output for the vectors, quaternions and radians? It's as simple as this, for *any* Magnum type:

```cpp
std::ostringstream out;
out << get<T>();
return out.str();
```

Besides minor formatting differences, I don't see why you would have to write all that code again.

Reply:
The to-strings were done this way originally (with vec3 and quats) to remove the commas between the elements (this function was being used to provide data in a csv-friendly format).  I figured that if folks wanted the magnum-based stream version, they could get it easily enough in the manner you described.

Reply:
@mosra Is there a flag for Magnum::Debug::Flag for NoComma?

Reply:
I was looking for that, but it looks like the stream functions [specify the comma explicitly ](https://github.com/mosra/magnum/blob/5f5cfb10946a344b9dab704cbf6922fc3ee5bbb0/src/Magnum/Math/RectangularMatrix.h#L682)

Reply:
Yeah the formatting is static because the intention is for it to be human-readable, not machine-consumable. Though in your case it's not CSV-friendly either due to the `[ ]`... :) 

I have a different idea for a generic machine-readable output -- all Math classes have a `.data()` accessor, which (in latest Magnum) provides a sized array reference. So then you could have a single function, such as `printCsvFriendly(Cr::Containers::ArrayView<const T>)`, which converts a numeric array into a space-delimited string, and then use it for any Math type.